### PR TITLE
feat: Add Markdown display widget to extensibility

### PR DESCRIPTION
**Pull request by mrCherry97**
**Description**

Changes proposed in this pull request:

- Add a new `Markdown` extensibility display widget that renders a string value (for example, a ConfigMap `data` entry) as formatted Markdown, instead of showing it in a read-only code editor.
- Render Markdown through the existing `marked-react` dependency. Any embedded raw HTML is rendered as escaped text and never executed (XSS-safe).
- Add theme-aware styles using SAP CSS variables, so the widget looks correct in both light and dark themes.
- Register the widget in the extensibility widget map and document it in the list-and-details widgets guide with a usage example.
- Add a Cypress component test covering headings, lists, the empty-value placeholder, and HTML escaping.

**Related issue(s)**

No tracking issue — small, self-contained widget addition.

**Definition of done**

- [x] The PR's title starts with one of the following prefixes:
  - feat: A new feature
  - fix: A bug fix
  - docs: Documentation only changes
  - refactor: A code change that neither fixes a bug nor adds a feature
  - test: Adding tests
  - revert: Revert commit
  - chore: Maintainance changes to the build process or auxiliary tools, libraries, workflows, etc.
- Related issues: not applicable — there is no tracking issue for this change.
- [x] Explain clearly why you created the PR and what changes it introduces
- [x] All necessary steps are delivered, for example, tests, documentation, merging

### CODE REVIEW

**Review by OliwiaGowor — APPROVED** [review]

File: docs/contributor/extensibility/50-list-and-details-widgets.md [hunk 1]
```diff
@@ -558,6 +559,28 @@ injections: |-
 
 <img src="./assets/display-widgets/FeaturedCard.png" alt="Example of a FeaturedCard widget">
 
+### `Markdown`
+
+The `Markdown` widget renders a string value as formatted Markdown. It is useful for displaying human-readable, richly formatted text stored as a plain string (for example, a `ConfigMap` **data** entry), instead of showing it in a read-only code editor.
```
Comment (grego952) [inline docs/contributor/extensibility/50-list-and-details-widgets.md]:
```suggestion
The `Markdown` widget renders a string value as formatted Markdown. It is useful for displaying human-readable, richly formatted text stored as a plain string (for example, a `ConfigMap` data entry), instead of showing it in a read-only code editor.
```

**Review by grego952 — APPROVED** [review]

**Comment by mrCherry97** [comment]
Bypassing the merge queue due to the flakiness.
